Bug: mkcert not install in ubuntu during setup

### Description

I expect all requirements for aidevops to be installed with setup.sh

### Steps to Reproduce

use ubuntu 24.04 and try to run aidevops init

### Expected Behavior

everything will work without error

### Error Output

```shell
aidevops init
Initialize AI DevOps in Project


[WARN] On protected branch 'main'

Options:
  1. Create worktree: chore/aidevops-init (recommended)
  2. Continue on main (commits directly to main)
  3. Cancel

Choice [1]: 1
[INFO] Creating worktree at /home/dave/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager-chore-aidevops-init...
Creating worktree with new branch 'chore/aidevops-init'...
HEAD is now at c1e5d6c chore: rename composer package from devstone/ultimate-multisite-plugin-and-theme-manager to ultimate-multisite/ultimate-multisite-plugin-and-theme-manager

Worktree created successfully!

Path: /home/dave/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager-chore-aidevops-init
Branch: chore/aidevops-init

To start working:
  cd /home/dave/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager-chore-aidevops-init

Or open in a new terminal/editor:
  code /home/dave/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager-chore-aidevops-init        # VS Code
  cursor /home/dave/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager-chore-aidevops-init      # Cursor
  opencode /home/dave/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager-chore-aidevops-init    # OpenCode

Localdev integration: auto-registering project 'ultimate-multisite-plugin-and-theme-manager'...
[INFO] localdev add ultimate-multisite-plugin-and-theme-manager (ultimate-multisite-plugin-and-theme-manager.local)

[INFO] Auto-assigning port from range 3100-3999...
[SUCCESS] Assigned port: 3100
[INFO] Generating mkcert wildcard cert for *.ultimate-multisite-plugin-and-theme-manager.local and ultimate-multisite-plugin-and-theme-manager.local...
/home/dave/.aidevops/agents/scripts/localdev-helper.sh: line 680: mkcert: command not found
[ERROR] mkcert failed to generate cert files
[INFO]   Expected: /home/dave/.local-ssl-certs/ultimate-multisite-plugin-and-theme-manager.local+1.pem
[INFO]   Expected: /home/dave/.local-ssl-certs/ultimate-multisite-plugin-and-theme-manager.local+1-key.pem
Localdev auto-registration failed (non-fatal)

[OK] Worktree created!
[INFO] Switching to: /home/dave/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager-chore-aidevops-init

[INFO] Project root: /home/dave/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager-chore-aidevops-init

[INFO] Creating .aidevops.json...
[OK] Created .aidevops.json
[OK] Created .agents/ directory
[OK] Created .agents/AGENTS.md
[OK] Created AGENTS.md
[INFO] Setting up planning files...
[OK] Created TODO.md
[OK] Created todo/PLANS.md
[INFO] Setting up database schema directories...
[OK] Created schemas/ directory
[OK] Created migrations/ directory
[OK] Created seeds/ directory
[INFO] Setting up Beads task graph...
[INFO] Initializing Beads database...
[WARN] Beads init failed - run manually: bd init
[INFO] Syncing tasks to Beads...
[OK] Tasks synced to Beads
dave@thinkserver:~/tgc.church/site/web/app/plugins/ultimate-multisite-plugin-and-theme-manager$ mkcert
Command 'mkcert' not found, but can be installed with:
sudo apt install mkcert
```

### aidevops Version

3.1.239

### Operating System

Ubuntu / Debian

### AI Assistant

OpenCode

### Third-party Tool Involvement

_No response_

### Additional Context

_No response_

### Pre-submission Checklist

- [x] I searched existing issues and this is not a duplicate
- [x] I am running the latest version of aidevops (or noted my version above)

--- a/.agents/scripts/localdev-helper.sh
+++ b/.agents/scripts/localdev-helper.sh
@@ -181,10 +181,8 @@ check_init_prerequisites() {
 	command -v docker >/dev/null 2>&1 || missing+=("docker")
 
 	# Try to auto-install mkcert if missing (GH#6415)
-	if ! command -v mkcert >/dev/null 2>&1; then
-		if ! ensure_mkcert; then
-			missing+=("mkcert")
-		fi
+	if ! command -v mkcert >/dev/null 2>&1 && ! ensure_mkcert; then
+		missing+=("mkcert")
 	fi
 
 	# dnsmasq: check brew installation or system-wide
@@ -777,11 +775,9 @@ generate_cert() {
 	local wildcard="*.${domain}"
 
 	# Ensure mkcert is available (auto-install if missing, GH#6415)
-	if ! command -v mkcert >/dev/null 2>&1; then
-		if ! ensure_mkcert; then
-			print_error "mkcert is required to generate SSL certificates"
-			return 1
-		fi
+	if ! command -v mkcert >/dev/null 2>&1 && ! ensure_mkcert; then
+		print_error "mkcert is required to generate SSL certificates"
+		return 1
 	fi
 
 	mkdir -p "$CERTS_DIR"

--- a/.agents/scripts/localdev-helper.sh
+++ b/.agents/scripts/localdev-helper.sh
@@ -81,7 +81,9 @@ cmd_init() {
 
 # Install mkcert if not present. Supports macOS (brew) and Linux (apt, dnf,
 # pacman, apk). On Linux, the apt package name is "mkcert" (available in
-# Ubuntu 20.04+ and Debian 11+). After installation, runs `mkcert -install`
+# Ubuntu 20.04+ and Debian 11+). libnss3-tools is required on Debian/Ubuntu
+# for mkcert to install the CA root into Firefox/Chrome trust stores.
+# After installation, runs `mkcert -install`
 # to create and trust the local CA root.
 # Returns: 0 if mkcert is available after this function, 1 if installation failed.
 ensure_mkcert() {
@@ -98,7 +100,7 @@ ensure_mkcert() {
 			installed=true
 		fi
 	elif command -v apt-get >/dev/null 2>&1; then
-		if sudo apt-get update -qq && sudo apt-get install -y -qq mkcert 2>/dev/null; then
+		if sudo apt-get update -qq && sudo apt-get install -y -qq mkcert libnss3-tools 2>/dev/null; then
 			installed=true
 		fi
 	elif command -v dnf >/dev/null 2>&1; then
@@ -119,7 +121,7 @@ ensure_mkcert() {
 		print_error "Failed to install mkcert automatically"
 		echo "  Manual install options:"
 		echo "    macOS:         brew install mkcert"
-		echo "    Ubuntu/Debian: sudo apt install mkcert"
+		echo "    Ubuntu/Debian: sudo apt install mkcert libnss3-tools"
 		echo "    Fedora:        sudo dnf install mkcert"
 		echo "    Arch:          sudo pacman -S mkcert"
 		echo "    Other:         https://github.com/FiloSottile/mkcert#installation"

--- a/.agents/scripts/localdev-helper.sh
+++ b/.agents/scripts/localdev-helper.sh
@@ -79,14 +79,82 @@ cmd_init() {
 	return 0
 }
 
+# Install mkcert if not present. Supports macOS (brew) and Linux (apt, dnf,
+# pacman, apk). On Linux, the apt package name is "mkcert" (available in
+# Ubuntu 20.04+ and Debian 11+). After installation, runs `mkcert -install`
+# to create and trust the local CA root.
+# Returns: 0 if mkcert is available after this function, 1 if installation failed.
+ensure_mkcert() {
+	if command -v mkcert >/dev/null 2>&1; then
+		return 0
+	fi
+
+	print_info "mkcert not found — attempting to install..."
+
+	local installed=false
+
+	if command -v brew >/dev/null 2>&1; then
+		if brew install mkcert 2>/dev/null; then
+			installed=true
+		fi
+	elif command -v apt-get >/dev/null 2>&1; then
+		if sudo apt-get update -qq && sudo apt-get install -y -qq mkcert 2>/dev/null; then
+			installed=true
+		fi
+	elif command -v dnf >/dev/null 2>&1; then
+		if sudo dnf install -y mkcert 2>/dev/null; then
+			installed=true
+		fi
+	elif command -v pacman >/dev/null 2>&1; then
+		if sudo pacman -S --noconfirm mkcert 2>/dev/null; then
+			installed=true
+		fi
+	elif command -v apk >/dev/null 2>&1; then
+		if sudo apk add mkcert 2>/dev/null; then
+			installed=true
+		fi
+	fi
+
+	if [[ "$installed" != "true" ]] || ! command -v mkcert >/dev/null 2>&1; then
+		print_error "Failed to install mkcert automatically"
+		echo "  Manual install options:"
+		echo "    macOS:         brew install mkcert"
+		echo "    Ubuntu/Debian: sudo apt install mkcert"
+		echo "    Fedora:        sudo dnf install mkcert"
+		echo "    Arch:          sudo pacman -S mkcert"
+		echo "    Other:         https://github.com/FiloSottile/mkcert#installation"
+		return 1
+	fi
+
+	print_success "mkcert installed"
+
+	# Install the local CA into the system trust store (one-time setup).
+	# This makes mkcert-generated certs trusted by browsers and curl.
+	print_info "Installing mkcert local CA root (may require sudo)..."
+	if mkcert -install 2>/dev/null; then
+		print_success "mkcert CA root installed and trusted"
+	else
+		print_warning "mkcert -install failed — certs will generate but browsers may not trust them"
+		echo "  Run manually: mkcert -install"
+	fi
+
+	return 0
+}
+
 # Check that required tools are installed
 check_init_prerequisites() {
 	local missing=()
 
 	command -v docker >/dev/null 2>&1 || missing+=("docker")
-	command -v mkcert >/dev/null 2>&1 || missing+=("mkcert")
 
-	# dnsmasq: check brew installation
+	# Try to auto-install mkcert if missing (GH#6415)
+	if ! command -v mkcert >/dev/null 2>&1; then
+		if ! ensure_mkcert; then
+			missing+=("mkcert")
+		fi
+	fi
+
+	# dnsmasq: check brew installation or system-wide
 	local brew_prefix
 	brew_prefix="$(detect_brew_prefix)"
 	if [[ -z "$brew_prefix" ]] || [[ ! -f "$brew_prefix/etc/dnsmasq.conf" ]]; then
@@ -97,7 +165,11 @@ check_init_prerequisites() {
 
 	if [[ ${#missing[@]} -gt 0 ]]; then
 		print_error "Missing required tools: ${missing[*]}"
-		echo "  Install: brew install ${missing[*]}"
+		echo "  Install:"
+		echo "    macOS:         brew install ${missing[*]}"
+		echo "    Ubuntu/Debian: sudo apt install ${missing[*]}"
+		echo "    Fedora:        sudo dnf install ${missing[*]}"
+		echo "    Arch:          sudo pacman -S ${missing[*]}"
 		exit 1
 	fi
 
@@ -670,6 +742,14 @@ generate_cert() {
 	local name="$1"
 	local domain="${name}.local"
 	local wildcard="*.${domain}"
+
+	# Ensure mkcert is available (auto-install if missing, GH#6415)
+	if ! command -v mkcert >/dev/null 2>&1; then
+		if ! ensure_mkcert; then
+			print_error "mkcert is required to generate SSL certificates"
+			return 1
+		fi
+	fi
 
 	mkdir -p "$CERTS_DIR"
 
@@ -2688,7 +2768,9 @@ cmd_help() {
 	echo "  Note: dnsmasq resolves .local for CLI tools only. Browsers need /etc/hosts"
 	echo "  entries (added automatically by 'add' command) due to macOS mDNS."
 	echo ""
-	echo "Requires: docker, mkcert, dnsmasq (brew install dnsmasq)"
+	echo "Requires: docker, mkcert, dnsmasq"
+	echo "  mkcert is auto-installed if missing (apt, dnf, pacman, brew)"
+	echo "  dnsmasq: brew install dnsmasq (macOS) / sudo apt install dnsmasq (Linux)"
 	echo "Requires: sudo (for /etc/hosts and dnsmasq restart)"
 	echo ""
 	echo "LocalWP coexistence:"

--- a/.agents/scripts/localdev-helper.sh
+++ b/.agents/scripts/localdev-helper.sh
@@ -83,6 +83,8 @@ cmd_init() {
 # pacman, apk). On Linux, the apt package name is "mkcert" (available in
 # Ubuntu 20.04+ and Debian 11+). libnss3-tools is required on Debian/Ubuntu
 # for mkcert to install the CA root into Firefox/Chrome trust stores.
+# Falls back to downloading the upstream binary from dl.filippo.io for
+# distributions without a supported package manager (x86_64, arm64, armv7l).
 # After installation, runs `mkcert -install`
 # to create and trust the local CA root.
 # Returns: 0 if mkcert is available after this function, 1 if installation failed.
@@ -114,6 +116,35 @@ ensure_mkcert() {
 	elif command -v apk >/dev/null 2>&1; then
 		if sudo apk add mkcert 2>/dev/null; then
 			installed=true
+		fi
+	fi
+
+	# Binary download fallback for distros without a supported package manager.
+	# Supports x86_64 (amd64), aarch64/arm64, and armv7l architectures.
+	if [[ "$installed" != "true" ]] && command -v curl >/dev/null 2>&1; then
+		local raw_arch
+		raw_arch=$(uname -m)
+		local arch=""
+		case "$raw_arch" in
+		x86_64) arch="amd64" ;;
+		aarch64 | arm64) arch="arm64" ;;
+		armv7l) arch="armv7l" ;;
+		*) arch="" ;;
+		esac
+
+		if [[ -n "$arch" ]]; then
+			print_info "Attempting binary download fallback for linux/$arch..."
+			local bin_dir="$HOME/.local/bin"
+			mkdir -p "$bin_dir"
+			local mkcert_url="https://dl.filippo.io/mkcert/latest?for=linux/$arch"
+			if curl -fsSL "$mkcert_url" -o "$bin_dir/mkcert" 2>/dev/null && chmod +x "$bin_dir/mkcert"; then
+				# Ensure ~/.local/bin is on PATH for this session
+				export PATH="$bin_dir:$PATH"
+				if command -v mkcert >/dev/null 2>&1; then
+					installed=true
+					print_success "mkcert installed via binary download to $bin_dir/mkcert"
+				fi
+			fi
 		fi
 	fi
 

